src/namespace.ts

```diff
@@ -59,4 +59,17 @@ export const namespaceActionDispatcher = (
   namespace: string,
   parentDispatcher: ActionDispatcher
 ): ActionDispatcher => (action) =>
-  parentDispatcher(_namespaceAction(namespace, action));
+  dispatchNamespaced(namespace, parentDispatcher, action);
+
+/**
+ * Add the namespace to the action and dispatch it.
+ *
+ * @param namespace Namespace to dispatch action with
+ * @param parentDispatcher Action dispatcher to dispatch action with
+ * @param action The action to namespace and dispatch
+ */
+export const dispatchNamespaced = (
```

Comment:
I think the interface is too complicated to use namespaces. Ideally the namespace should IMO just be an optional argument of `dispatchAction`, basically `dispatchAction(action: Action, namespace?: Namespace)` with `Namespace` being an enum. Accordingly in the creating stream functions the namespace would be the last optional argument too. By default the namespace is null.

Reply:
I agree, this is too complicated. I looked into something like an optional argument to `dispatchAction`, but I don't remember why I didn't go for it.

I will take a look at it again, and probably create a PR which brings the `action$` and `dispatchAction` into RxBeach.

Reply:
New PR is up in #234